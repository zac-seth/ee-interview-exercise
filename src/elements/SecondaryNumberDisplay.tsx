import { FC } from "react";
import { NumberDisplay, NumberDisplayMode } from "./NumberDisplay.style";

export type SecondaryNumberDisplayProps = {
  value: string;
};

export const SecondaryNumberDisplay: FC<SecondaryNumberDisplayProps> = ({
  value
}) => <NumberDisplay mode={NumberDisplayMode.SECONDARY}>{value}</NumberDisplay>;
