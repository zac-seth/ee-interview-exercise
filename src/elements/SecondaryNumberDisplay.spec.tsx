import { render, screen } from '@testing-library/react';

import { SecondaryNumberDisplay } from './SecondaryNumberDisplay';

describe(`SecondaryNumberDisplay`, () => {
  it(`Should render the component`, () => {
    render(<SecondaryNumberDisplay value='10 + 15 &times;' />);

    expect(screen.getByText(`10 + 15 Ã—`)).toMatchSnapshot();
  });
});
