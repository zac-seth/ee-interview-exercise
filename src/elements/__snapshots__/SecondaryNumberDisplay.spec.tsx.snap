// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecondaryNumberDisplay Should render the component 1`] = `
.c0 {
  text-shadow: 0 0 5px #FFF,0 0 15px #B6FF00;
  color: #FFF;
  margin: 0;
  padding: 0;
  font-size: 3.2rem;
  line-height: 3.2rem;
}

<p
  class="c0"
  mode="secondary"
>
  10 + 15 Ã—
</p>
`;
